detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/GoModCommandExecutor.java

```diff
@@ -0,0 +1,27 @@
+package com.synopsys.integration.detectable.detectables.go.gomod;
+
+import java.io.File;
+import java.util.List;
+
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableOutput;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableRunner;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableRunnerException;
+
+public class GoModCommandExecutor {
```

Comment:
This class isn't doing something particularly complex or GoMod specific.  I would have GoListUJsonGenerator call the ExecutableRunner directly and save the complication of an additional class.

Reply:
The only reason I pulled it out is because otherwise you'd have that execute wrapper method in every class that runs a go mod command (and it didn't appear that any other detectors used that wrapper method).  If you think that would be preferable, I can do that.

Reply:
I'd be fine adding a method to ExecutableOutput called `throwOnNonZeroExitCode(String message)` so that 
`
        ExecutableOutput output = executableRunner.execute(directory, goExe, arguments);

        if (output.getReturnCode() == 0) {
            return output.getStandardOutputAsList();
        } else {
            throw new DetectableException(failureMessage + output.getReturnCode());
        }
`
would become 
`
        ExecutableOutput output = executableRunner.execute(directory, goExe, arguments);
        output.throwOnNonZeroExitCode(failureMessage);
        return output.getStandardOutputAsList();
`
And if that's the only thing in this class, I have no problem with GoListUJsonGenerator and GoModGraphTransformer both having a method like this. In reality I think only GoListUJsonGenerator would benefit from putting that code in a method.